lib/pure/sugar.nim

```diff
@@ -198,3 +198,42 @@ macro dump*(x: typed): untyped =
   let r = quote do:
     debugEcho `s`, " = ", `x`
   return r
+
+# TODO: consider exporting this in macros.nim
+proc replaceNodes(ast: NimNode): NimNode =
```

Comment:
This should be called ``freshIdentNodes`` or similar.

Reply:
done